test/e2e/build_test.go

```diff
@@ -300,6 +300,11 @@ var _ = Describe("Podman build", func() {
 	})
 
 	It("podman build --http_proxy flag", func() {
+		if env, found := os.LookupEnv("http_proxy"); found {
+			defer os.Setenv("http_proxy", env)
```

Comment:
I am not sure it does what you want.  Here, the env variable is already set.  The defer will set it again but it has not been unset as far as I can see.

Reply:
https://github.com/containers/podman/blob/8b87665f235b84ac9c3795970d51c2ca898ffe00/test/e2e/build_test.go#L308

The `http_proxy` value is overwritten to `1.2.3.4` due to `os.Setenv("http_proxy", "1.2.3.4") `.
This value needs to revert the original value at this time.

`defer os.Setenv("http_proxy", env)` reverts the `http_proxy` to original value.